src/query/api/v1/handler/prometheus/tags/tag_values.go

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package tags
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/m3db/m3/src/query/api/v1/handler"
+	"github.com/m3db/m3/src/query/storage"
+	"github.com/m3db/m3/src/query/util/logging"
+	"github.com/m3db/m3/src/x/net/http"
+
+	"go.uber.org/zap"
+)
+
+const (
+	// TagValuesURL is the url for tag values.
+	TagValuesURL = handler.RoutePrefixV1 + "/:" + nameReplace + "/values"
+
+	// TagValuesHTTPMethod is the HTTP method used with this resource.
+	TagValuesHTTPMethod = http.MethodGet
+)
+
+// TagValuesHandler represents a handler for search tags endpoint.
+type TagValuesHandler struct {
+	storage storage.Storage
+}
+
+// TagValuesResponse is the response that gets returned to the user
+type TagValuesResponse struct {
+	Results storage.CompleteTagsResult `json:"results,omitempty"`
+}
+
+// NewTagValuesHandler returns a new instance of handler.
+func NewTagValuesHandler(
+	storage storage.Storage,
+) http.Handler {
+	return &CompleteTagsHandler{
+		storage: storage,
+	}
+}
+
+func (h *TagValuesHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	ctx := context.WithValue(r.Context(), handler.HeaderKey, r.Header)
+	logger := logging.WithContext(ctx)
+	w.Header().Set("Content-Type", "application/json")
```

Comment:
move this into a common place ? All the handlers seem to be setting these headers

Reply:
+1, we could make this middleware.